src/main/kotlin/com/kostas/kafkaStreams/config/Initializer.kt

```diff
@@ -0,0 +1,20 @@
+package com.kostas.kafkaStreams.config
+
+import com.kostas.kafkaStreams.ListingProducer
+import com.kostas.kafkaStreams.model.ListingSummary
+import com.kostas.kafkaStreams.util.ResourceFileReader
+import org.springframework.context.annotation.Configuration
+import javax.annotation.PostConstruct
+
+@Configuration
+class Initializer(
+    private val listingProducer: ListingProducer,
+    private val resources: ResourceFileReader
+) {
+
+    @PostConstruct
```

Comment:
Can also achieve this with an `init {...}` block

Reply:
Any idea what the difference is? Is one more appropriate?

Reply:
I don’t think there’s a major difference. Both would work. I think this way this block will be called after the rest of configuration and all components have been created first.